### ISSUE TITLE
Long running (resque) processes trigger DRb Conn Error with Broken Pipe

### ISSUE THREAD

**Issue by serioushaircut** [appsignal/appsignal#38]
Hi Appsignal,

we use version 0.11.9 with Resque Plugin and a Worker runs sometimes longer than 45min. 
The following error is reproducible:

```
Exception DRb::DRbConnError 
Error Broken pipe
/usr/lib/ruby/2.1.0/drb/drb.rb:611:in `write'
/usr/lib/ruby/2.1.0/drb/drb.rb:611:in `send_request'
/usr/lib/ruby/2.1.0/drb/drb.rb:921:in `send_request'
/usr/lib/ruby/2.1.0/drb/drb.rb:1221:in `send_message'
/usr/lib/ruby/2.1.0/drb/drb.rb:1110:in `block (2 levels) in method_missing'
/usr/lib/ruby/2.1.0/drb/drb.rb:1197:in `open'
/usr/lib/ruby/2.1.0/drb/drb.rb:1109:in `block in method_missing'
/usr/lib/ruby/2.1.0/drb/drb.rb:1128:in `with_friend'
/usr/lib/ruby/2.1.0/drb/drb.rb:1108:in `method_missing'
/app/.bundle/ruby/2.1.0/gems/appsignal-0.11.9/lib/appsignal/ipc.rb:59:in `enqueue'
/app/.bundle/ruby/2.1.0/gems/appsignal-0.11.9/lib/appsignal/transaction.rb:160:in `complete!'
/app/.bundle/ruby/2.1.0/gems/appsignal-0.11.9/lib/appsignal/transaction.rb:26:in `complete_current!'
/app/.bundle/ruby/2.1.0/gems/appsignal-0.11.9/lib/appsignal.rb:92:in `monitor_transaction'
/app/.bundle/ruby/2.1.0/gems/appsignal-0.11.9/lib/appsignal/integrations/resque.rb:8:in `around_perform_resque_plugin'
```

We think that looks like, that on completing the job finally, appsignal tries to enqueue the transaction into the transaction Drb server file 'drbunix:...' stream which is no longer opened for writing.

Is that anything we can just configure anywhere in case we know our job might run quite some time? Or is it a bug?

Cheers,
Bastian

**Comment by thijsc** [comment]
We made a fix for this in #39. Could you try that and let us know if it works well for you?

**Comment by serioushaircut** [comment]
I close this issue as it does not state the error correctly (runtime is not the cause, probably the transaction object size - see #39 for more details)
